alacritty/src/window.rs

```diff
@@ -144,141 +129,125 @@ fn create_gl_window(
 
         // Text cursor
-        window.set_cursor(MouseCursor::Text);
+        window.set_cursor_icon(CursorIcon::Text);
 
         // Set OpenGL symbol loader. This call MUST be after window.make_current on windows.
-        gl::load_with(|symbol| windowed_context.get_proc_address(symbol) as *const _);
-
-        let window = Window {
-            event_loop,
-            current_mouse_cursor: MouseCursor::Default,
-            windowed_context,
-            mouse_visible: true,
-            is_focused: false,
-        };
+        gl::load_with(|symbol| context.get_proc_address(symbol) as *const _);
 
-        window.run_os_extensions();
+        let window =
+            Window { window, mouse_visible: true, current_mouse_cursor: CursorIcon::Default };
 
-        Ok(window)
-    }
+        window.run_os_extensions();
 
-    /// Get some properties about the device
-    ///
-    /// Some window properties are provided since subsystems like font
-    /// rasterization depend on DPI and scale factor.
-    pub fn device_properties(&self) -> DeviceProperties {
-        DeviceProperties { scale_factor: self.window().get_hidpi_factor() }
+        Ok(WindowedContext { window, context })
     }
+}
 
-    pub fn inner_size_pixels(&self) -> Option<LogicalSize> {
-        self.window().get_inner_size()
-    }
+/// A window which can be used for displaying the terminal
+///
+/// Wraps the underlying windowing library to provide a stable API in Alacritty
+pub struct Window {
+    window: GlutinWindow,
+    mouse_visible: bool,
+    current_mouse_cursor: CursorIcon,
+}
 
+impl Window {
     pub fn set_inner_size(&mut self, size: LogicalSize) {
-        self.window().set_inner_size(size);
-    }
-
-    #[inline]
-    pub fn hidpi_factor(&self) -> f64 {
-        self.window().get_hidpi_factor()
+        self.window.set_inner_size(size);
     }
 
-    #[inline]
-    pub fn create_window_proxy(&self) -> Proxy {
-        Proxy { inner: self.event_loop.create_proxy() }
-    }
-
-    #[inline]
-    pub fn swap_buffers(&self) -> Result<()> {
-        self.windowed_context.swap_buffers().map_err(From::from)
+    pub fn inner_size(&self) -> LogicalSize {
+        self.window.inner_size()
     }
 
-    /// Poll for any available events
-    #[inline]
-    pub fn poll_events<F>(&mut self, func: F)
-    where
-        F: FnMut(Event),
-    {
-        self.event_loop.poll_events(func);
-    }
-
-    #[inline]
-    pub fn resize(&self, size: PhysicalSize) {
-        self.windowed_context.resize(size);
+    pub fn hidpi_factor(&self) -> f64 {
+        self.window.hidpi_factor()
     }
 
     /// Show window
     #[inline]
-    pub fn show(&self) {
-        self.window().show();
-    }
-
-    /// Block waiting for events
-    #[inline]
-    pub fn wait_events<F>(&mut self, func: F)
-    where
-        F: FnMut(Event) -> ControlFlow,
-    {
-        self.event_loop.run_forever(func);
+    pub fn set_visible(&self, visibility: bool) {
```

Comment:
It's not `Show window` to be true.

Reply:
Good point, just going to remove the comment because I don't think adding a comment on `Window::set_visible` will do us any good.